src/lib/party.ts

```diff
@@ -719,11 +729,12 @@ export class PartyBalance {
   }
 }
 
-export class Party<P> {
+type Predicate = undefined | UInt32 | AccountPredicate;
+export class Party {
   body: Body;
-  predicate: P;
+  predicate: Predicate;
```

Comment:
Can you explain why you moved away from the generic type here?

Reply:
I don't remember, I can only guess that keeping track of the type parameter in places that had nothing to do with the account predicate became too annoying. The union type we have now makes distinguishing between different variants more localized

Reply:
Seems fair enough 👍